frontend/src/charts/bar.ts

```diff
@@ -56,26 +56,66 @@ const bar_validator = array(
   })
 );
 
+/** Calculate the currencies to use for the chart. */
+function calculateCurrenciesToShow(
+  data: {
+    budgets: Record<string, number>;
+    balance: Record<string, number>;
+  }[],
+  operatingCurrencies: string[]
+): string[] {
+  // Count the usage of each currency in the data.
+  const inData = data.reduce((map, interval) => {
```

Comment:
This could be handled with a one-liner with the `rollup` function from `d3-array` (https://devdocs.io/d3~7/d3-array#rollup)